frontend/src/app/project/template-variables.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import {ValueInfo} from "../form/value-input/value-info";
+import {TemplateVariable} from "./project";
+
+@Injectable()
+export class TemplateVariablesService {
+
+  constructor() { }
+
+  public createValueInfo(templateVariable: TemplateVariable, selectedValue: string): ValueInfo {
```

Comment:
This method does not justify a service as it's just used to create an object from two parameters. I would prefer `ValueInfo` to be a class and use those parameters in the constructor (or a static method).